src/AddressSpace.h

```diff
@@ -242,6 +245,9 @@ class AddressSpace : public HasTaskSet {
     // equal to 'map'.
     const KernelMapping recorded_map;
     const EmuFile::shr_ptr emu_file;
+    // If this mapping has been mapped into the local address space, the address
+    // is recorded here, NULL otherwise.
+    void* local_addr;
```

Comment:
Please make it explicit that this points to the first byte of this Mapping (once that's true!). And also note that responsibility for keeping local_addr alive at least as long as the Mapping rests with whoever mapped it in the first place.